src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -131,25 +167,26 @@ public synchronized InputStepExecution getExecution(String id) {
         return null;
     }
 
-    public synchronized List<InputStepExecution> getExecutions() {
+    public synchronized List<InputStepExecution> getExecutions() throws InterruptedException, TimeoutException {
         loadExecutions();
         return new ArrayList<InputStepExecution>(executions);
     }
 
     /**
      * Called when {@link InputStepExecution} is completed to remove it from the active input list.
      */
-    public synchronized void remove(InputStepExecution exec) throws IOException {
+    public synchronized void remove(InputStepExecution exec) throws IOException, InterruptedException, TimeoutException {
         loadExecutions();
         executions.remove(exec);
         ids.remove(exec.getId());
+        inputDefinitions.remove(exec.getId());
         run.save();
     }
 
     /**
      * Bind steps just by their ID names.
      */
-    public InputStepExecution getDynamic(String token) {
+    public InputStepExecution getDynamic(String token) throws InterruptedException, TimeoutException {
```

Comment:
I suppose you would want to deprecate this and update the UI code in this plugin correspondingly. Which would also be a good time to fix [JENKINS-34509](https://issues.jenkins-ci.org/browse/JENKINS-34509).